lib/job/arSearchResultExportCsvJob.class.php

```diff
@@ -0,0 +1,235 @@
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * A worker to, given the HTTP GET parameters sent to advanced search,
+ * replicate the search and export the resulting decriptions to CSV.
+ *
+ * @package    symfony
+ * @subpackage jobs
+ */
+
+class arSearchResultExportCsvJob extends arBaseJob
+{
+  /**
+   * @see arBaseJob::$requiredParameters
+   */
+  protected $extraRequiredParameters = array('params');  // search params
+  protected $downloadFileExtension = 'zip';
+
+  protected $searchParams;      // key/value array of search terms
+  protected $search;            // arElasticSearchPluginQuery instance
+
+  protected $archivalStandard;  // which CSV export configuration to use: either "rad" or "isad"
+
+  public function runJob($parameters)
+  {
+    $this->searchParams = $parameters['params'];
+    $this->search = new arElasticSearchPluginQuery();
+    $this->addCriteriaBasedOnSearchParameters();
+
+    // Sort by ID so parents can import properly from resulting export file
+    $this->search->query->setSort(array('lft' => 'asc'));
+
+    $this->archivalStandard = QubitSetting::getByNameAndScope('informationobject', 'default_template');
+
+    // Create temp directory in which CSV export files will be written
+    $tempPath = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'search_export_'. $this->job->id;
+    mkdir($tempPath);
+
+    // Export CSV to temp directory
+    $this->info('Starting export to '. $tempPath .'.');
+    $itemsExported = $this->exportResults($tempPath);
+    $this->info('Exported '. $itemsExported .' descriptions.');
+
+    // Compress CSV export files as a ZIP archive
+    $this->info('Creating ZIP file '. $this->getDownloadFilePath() .'.');
+    $success = $this->createZipForDownload($tempPath);
+
+    if ($success !== true)
+    {
+      $this->error('Failed to create ZIP file.');
+      return false;
+    }
+
+    // Mark job as complete and set download path
+    $this->info('Export and archiving complete.');
+    $this->job->setStatusCompleted();
+    $this->job->downloadPath = $this->getDownloadRelativeFilePath();
+    $this->job->save();
+
+    return true;
+  }
+
+  /**
+   * Add criteria to query based on parameters.
+   *
+   * @return void
+   */
+  protected function addCriteriaBasedOnSearchParameters()
+  {
+    // Add criteria for main search fields
+    if (null !== $criterias = $this->parseQuery())
+    {
+      $this->search->queryBool->addMust($criterias);
+    }
+
+    // Add criteria fo secondary search fields
+    foreach (SearchAdvancedAction::$NAMES as $name)
+    {
+      if (
```

Comment:
Minor nitpick but in most of AtoM we don't do newlines after the (, and the ) usually isn't on its own line either.